### PR TITLE
Fix StringName Inequality Operators

### LINKED ISSUE #36511: "Duplicate key found in Dictionary literal" false error in 4.0 branch
**Godot version:**
4.0.dev.custom_build. 5e181ac9f

**Issue description:**
When I use this code
```
var rr = {x=5, y=2, type=55}
```
then I got this error 
```
SCRIPT ERROR: Parse Error: Duplicate key found in Dictionary literal
```
This worked fine in Godot 3.2

### PR DESCRIPTION
**Pull request by nathanfranke**
Fixes #36511

This one got me for a good 4 hours!

image

### CODE REVIEW

File: core/variant_op.cpp [hunk 1]
```diff
@@ -678,6 +678,7 @@ void Variant::evaluate(const Operator &p_op, const Variant &p_a,
 			DEFAULT_OP_LOCALMEM(math, OP_LESS, VECTOR2I, <, Vector2i);
 			DEFAULT_OP_LOCALMEM(math, OP_LESS, VECTOR3, <, Vector3);
 			DEFAULT_OP_LOCALMEM(math, OP_LESS, VECTOR3I, <, Vector3i);
+			DEFAULT_OP_LOCALMEM(math, OP_LESS, STRING_NAME, <, StringName);
```
Comment (akien-mga) [inline core/variant_op.cpp]: I see that there's `DEFAULT_OP_STR*` macros which are supposed to handle `StringName` too - maybe they should be used here?

I also wonder whether `NodePath` has the same issue.

Comment (KoBeWi) [reply core/variant_op.cpp]: NodePath works fine.

Comment (reduz) [reply core/variant_op.cpp]: SringNames are not necesarily strings, so technically only ==, != and should be implemented imo,and < for sorting.

**Comment by muiroc** [comment]
It fixes the script error but there is something weird going on, I gave it a try, applying over 1e255532e90fa35148167e4c153ad39f77010a04
try running this script
```gdscript
tool
extends EditorScript

func test(opts):
	#var should_be_true = opts.should_be_true # with this:  # Invalid get index 'should_be_true' (on base: 'Dictionary').
	var should_be_true = opts.get("should_be_true", false)
	print("should_be_true: ", should_be_true)
	
func _run():
	test({ should_be_true = true })  # outputs: should_be_true: False
	test({ "should_be_true": true }) # outputs: should_be_true: True

```
so the first dictionary literal syntax doesn't define the properties

**Comment by AndreaCatania** [comment]
I've tested this to check if this problem is solved too: https://github.com/godotengine/godot/issues/36900 but it's not.

Any idea what could cause it?

**Comment by vnen** [comment]
I believe this is now obsolete given the issue was fixed in another way and this code was refactored. Thanks for the contribution anyway!

**Pull request closed without merging**